docs/src/checklists.md

```diff
@@ -0,0 +1,38 @@
+# Checklists
+
+The purpose of this page is to collate a series of checklists for commonly
+performed changes to the source code of Documenter.
+
+In each case, copy the checklist into the description of the pull request.
+
+## Making a release
+
+In preparation for a release, use the following checklist.
+
+````
+## Pre-release
+
+ - [ ] Change the version number in `Project.toml`
+        * If the release is breaking, increment MAJOR
+        * If the release adds a new user-visible feature, increment MINOR
+        * Otherwise (bug-fixes, documentation improvements), increment PATCH
+ - [ ] Update `CHANGELOG.md`, following the existing style (in particular, make
+       sure that the change log for this version has the correct version number
+       and date).
+ - [ ] Run `make changelog`, to make sure that all the issue references in
+       `CHANGELOG.md` are up to date.
+ - [ ] Check that the commit messages in this PR do not contain `[ci skip]`
+
+## The release
+
+ - [ ] After merging this pull request, tag the release. There are two options for this:
```

Comment:
```suggestion
 - [ ] After merging the pull request, tag the release. There are two options for this:
```

I suppose tagging *after* the pull request is merged is a smoother workflow when everything goes well, but there's still the small chance that the release will fail for some reason. If we're okay with amending the release commit (which is now on `master`) in that case, force-pushing it, and retriggering the release, it wouldn't be a problem. Failures should be very rare, especially if we make sure that we have something in the CI that checks that `compat` bounds are valid (or whatever other requirements the Registrator has for auto-merging an update), so maybe this isn't really anything to worry about